pkg/controller/pxc/upgrade.go

```diff
@@ -364,9 +368,37 @@ func getPodOrderInSts(stsName string, podName string) (int, error) {
 	return strconv.Atoi(podName[len(stsName)+1:])
 }
 
+func (r *ReconcilePerconaXtraDBCluster) waitHostgroups(cr *api.PerconaXtraDBCluster, sfsName string, pod *corev1.Pod, waitLimit int, logger logr.Logger) error {
+	if cr.Spec.ProxySQL == nil || cr.Spec.ProxySQL != nil && !cr.Spec.ProxySQL.Enabled {
+		return nil
+	}
+
+	database, err := r.proxyDB(cr)
+	if err != nil {
+		return errors.Wrap(err, "failed to get proxySQL db")
+	}
+
+	defer database.Close()
+
+	podNamePrefix := fmt.Sprintf("%s.%s.%s", pod.Name, sfsName, cr.Namespace)
+
+	return retry(time.Second*10, time.Duration(waitLimit)*time.Second,
+		func() (bool, error) {
+			present, err := database.PresentInHostgroups(podNamePrefix)
+			if err != nil && err != queries.ErrNotFound {
+				return false, errors.Wrap(err, "failed to get hostgroup status")
+			}
+			if !present {
+				return false, nil
+			}
+
+			logger.Info("pod present in hostgroups", "pod name", pod.Name)
+			return true, nil
+		})
+}
+
 func (r *ReconcilePerconaXtraDBCluster) waitUntilOnline(cr *api.PerconaXtraDBCluster, sfsName string, pod *corev1.Pod, waitLimit int, logger logr.Logger) error {
-	if cr.Spec.HAProxy != nil && cr.Spec.HAProxy.Enabled {
-		time.Sleep(5 * time.Second)
+	if cr.Spec.ProxySQL == nil || cr.Spec.ProxySQL != nil && !cr.Spec.ProxySQL.Enabled {
```

Comment:
lets keep it simple
```suggestion
	if cr.Spec.ProxySQL == nil || !cr.Spec.ProxySQL.Enabled {
```

Reply:
Resolved in 2ed5502e4e0199e41891f464fdc15aa1cbe27ae4